client/my-sites/earn/home.tsx

```diff
@@ -20,58 +19,80 @@ import PromoSection, { Props as PromoSectionProps } from 'calypso/components/pro
 import { CtaButton } from 'calypso/components/promo-section/promo-card/cta';
 import wp from 'calypso/lib/wp';
 import { isEligibleForProPlan } from 'calypso/my-sites/plans-comparison';
+import { useDispatch, useSelector } from 'calypso/state';
 import { bumpStat, composeAnalytics, recordTracksEvent } from 'calypso/state/analytics/actions';
 import { getEarningsWithDefaultsForSiteId } from 'calypso/state/memberships/earnings/selectors';
+import { getConnectedAccountIdForSiteId } from 'calypso/state/memberships/settings/selectors';
 import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
 import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import siteHasWordAds from 'calypso/state/selectors/site-has-wordads';
 import { getSitePlanSlug } from 'calypso/state/sites/plans/selectors';
 import { isJetpackSite } from 'calypso/state/sites/selectors';
-import getSiteBySlug from 'calypso/state/sites/selectors/get-site-by-slug';
-import { IAppState } from 'calypso/state/types';
-import { getSelectedSiteSlug, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
 import { isRequestingWordAdsApprovalForSite } from 'calypso/state/wordads/approve/selectors';
 import CommissionFees from './components/commission-fees';
 import type { Image } from 'calypso/components/promo-section/promo-card/index';
-import type { SiteSlug } from 'calypso/types';
+
 import './style.scss';
 
-interface ConnectedProps {
-	siteId: number;
-	selectedSiteSlug: SiteSlug | null;
-	isNonAtomicJetpack: boolean;
-	isLoading: boolean;
-	hasSimplePayments: boolean;
-	hasWordAdsFeature: boolean;
-	hasConnectedAccount: boolean | null;
-	hasSetupAds: boolean;
-	eligibleForProPlan?: boolean;
-	trackUpgrade: ( plan: string, feature: string ) => void;
-	trackLearnLink: ( feature: string ) => void;
-	trackCtaButton: ( feature: string ) => void;
-	isUserAdmin?: boolean;
-	commission: number | null;
-}
-
-const Home: FunctionComponent< ConnectedProps > = ( {
-	commission,
-	siteId,
-	selectedSiteSlug,
-	isNonAtomicJetpack,
-	isUserAdmin,
-	isLoading,
-	hasSimplePayments,
-	hasConnectedAccount,
-	hasSetupAds,
-	eligibleForProPlan,
-	trackUpgrade,
-	trackLearnLink,
-	trackCtaButton,
-	hasWordAdsFeature,
-} ) => {
+const Home = () => {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 	const [ peerReferralLink, setPeerReferralLink ] = useState( '' );
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
```

Comment:
There are also `getSelectedSiteId` and `getSelectedSiteSlug` selectors — since we just use slug+ID here, could be simpler to just use those? Not a blocker for merging and this is subjective. :-)